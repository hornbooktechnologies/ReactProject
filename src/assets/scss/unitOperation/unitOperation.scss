@import "../_color";
@import "../_variables";

.bg-grey-blue-40-a10 {
  background-color: rgba($grey_blue_40, 0.1);
}
.color-grey-blue-60 {
  color: $grey_blue_60;
}
.min-w-1190 {
  min-width: 1190px;
}
.m-w-110 {
  min-width: 110px;
}
.max-height-500 {
  max-height: $val10 * 50;
}
.font-weight-500 {
  font-weight: 500;
}
// uk Table start
.uk-table {
  &.uk-table-divider {
    th,
    td {
      border: 1px $solid $grey_blue_40;
      text-align: left;
      padding: $val10 $val10 - 2;
      color: $white;
      font-size: $val15 - 1;
      .icn {
        position: $relative;
        top: 2px;
        &.icn-right {
          float: $right;
        }
      }
    }
    th {
      background-color: rgba($white, 0.1);
      font-size: $val15 + 1;
      vertical-align: $middle;
      text-transform: $inherit;
      font-weight: 500;
      text-align: $center;
    }
  }
  &.uk-text-center {
    th,
    td {
      text-align: $center;
    }
  }
  &.uk-text-right {
    th,
    td {
      text-align: $right;
    }
  }
  &.uk-table-sm {
    th {
      font-size: $val15 - 2 !important;
    }
    td {
      font-size: $val15 - 3 !important;
    }
  }
  &.uk-table-middle {
    th,
    td {
      vertical-align: $middle !important;
    }
  }
}
.min-w-86 {
  min-width: 86px;
  width: fit-content;
  margin-top: $val0;
}

@media (max-width: 767px) {
  .media-mt-30 {
    margin-top: 3 * $val10;
  }
}

