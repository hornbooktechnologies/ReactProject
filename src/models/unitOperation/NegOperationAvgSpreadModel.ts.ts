export type NegOperationAvgSpreadModel = {negOperationAvrgSpread:AverageSpreadCardModel[]}
export type NegativeOperationAvgSpreadModel = AverageSpreadCardModel[]

export type AverageSpreadCardModel = {
  title: string,
  Suffix: string,
  plan: AverageSpreadRecordModel
  actual_forcast?: AverageSpreadRecordModel
  yearStartToPresent?: AverageSpreadRecordModel
}

export type AverageSpreadRecordModel = {
  header?:string | undefined
  previousValue?: number | undefined | string
  suffix?: string 
  value?: number | undefined | string
}
