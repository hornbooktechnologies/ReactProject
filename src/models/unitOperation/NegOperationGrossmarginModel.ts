  export type NegGrossmarginTableModel = negativeGrossMargin
  export type NegativeGrossMarginTableModel = negativeGrossMargin
  export type NegativeGrossMarginModel = {negativeGrossMargin:negativeGrossMargin}
  export type negativeGrossMargin = {
    actualOrForcastAnnual:number|null|undefined|string
    actualOrForcastEnd:number|null|undefined|string
    actualOrForcastPresent:number|null|undefined|string
    impactOnEBITDAAnnual:number|null|undefined|string
    impactOnEBITDAEnd:number|null|undefined|string
    impactOnEBITDAPresent:number|null|undefined|string
    planAnnual:number|null|undefined|string
    planEnd:number|null|undefined|string
    planPresent:number|null|undefined|string
    previousActualOrForcastAnnual:number|null|undefined|string
    previousActualOrForcastEnd:number|null|undefined|string
    previousActualOrForcastPresent:number|null|undefined|string
    previousImpactOnEBITDAAnnual:number|null|undefined|string
    previousImpactOnEBITDAEnd:number|null|undefined|string
    previousImpactOnEBITDAPresent:number|null|undefined|string
    previousPlanAnnual:number|null|undefined|string
    previousPlanEnd:number|null|undefined|string
    previousPlanPresent:number|null|undefined|string
    customClassTop: string|undefined
    customClassEnd: string|undefined
    prefix:string|undefined
    suffix:string|undefined
}
