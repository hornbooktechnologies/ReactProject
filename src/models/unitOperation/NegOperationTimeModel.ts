export type NegOperationTimeModel =  {
  header: string ,
  plan:number | string,
  actualOrForecast:number | string,
  previousPlan?:number | undefined | null | string,
  previousActualOrForecast?:number | undefined | null | string,
  suffix?:string | undefined | null,
  titleX?:string | undefined | null,
  titleY?:string | undefined | null,
}
export type NegativeOperationTimeModel =  NegOperationTimeModel[]
export type NegOperationTimeDonutChartModel = {negOperationTime :NegOperationTimeModel[] }

