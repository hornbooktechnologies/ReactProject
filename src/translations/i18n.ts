import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
