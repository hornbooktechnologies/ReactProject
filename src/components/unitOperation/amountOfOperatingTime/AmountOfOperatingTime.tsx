import React from "react";
import { useTranslation } from 'react-i18next';
import { NegOperationTimeDonutChartModel } from "../../../models/unitOperation/NegOperationTimeModel";
import { AppConstant, ChartType } from "../../../utils/AppConstants";
import { GetDonutChartSeriesValWithDiff, GetStrongRedDonutchart } from "../../../utils/DonutChartUtils";
import DonutChartLarge from '../../common/customPie/DonutChartLarge';

const AmountOfOperatingTime = ({negOperationTime}:NegOperationTimeDonutChartModel ) => {
  const { i18n } = useTranslation();
  let annualData = negOperationTime && negOperationTime[0]
  let startTopresentData =  negOperationTime && negOperationTime[1]
  let presentToendData =  negOperationTime && negOperationTime[2]
  const pieChartRenderSize = 190;
  const opexChartType = ChartType.COST;
  const unitOperationChartCustomData = GetStrongRedDonutchart()
  const annualCustomData = GetDonutChartSeriesValWithDiff(i18n,unitOperationChartCustomData,annualData)
  const startToPresentCustomData = GetDonutChartSeriesValWithDiff(i18n,unitOperationChartCustomData,startTopresentData)
  const presentToEndCustomData = GetDonutChartSeriesValWithDiff(i18n,unitOperationChartCustomData,presentToendData)
 
  return (
    <>
      <div>
        <h5 className="mb-24 mt-24 font-18 uk-text-medium">
          {i18n.t('UNIT_OPERATION.AMOUNT_OPERATING_SPREAD')}
        </h5>
        <div
          className="uk-grid uk-child-width-1-3@s uk-child-width-1-3@m"
          uk-grid="true"
        >
          <div className="uk-first-column">
            <div className="uk-card">
            <h6 className="font-16">{annualData?.header}</h6>
               <DonutChartLarge
                data={annualCustomData}
                width={pieChartRenderSize}
                height={pieChartRenderSize}
                chartType={opexChartType}
                chartSection={AppConstant.OPERATING_TIME}
                removeTextTop={true}
              />
            </div>
          </div>
          <div className="media-mt-30 uk-first-column">
            <div className="uk-card">
            <h6 className="font-16">{startToPresentCustomData?.header}</h6>
              <DonutChartLarge
                data={startToPresentCustomData}
                width={pieChartRenderSize}
                height={pieChartRenderSize}
                chartType={opexChartType}
                chartSection={AppConstant.OPERATING_TIME}
                removeTextTop={true}
              />
            </div>
          </div> 
          <div className=" media-mt-30 uk-first-column">
            <div className="uk-card">
            <h6 className="font-16">{presentToEndCustomData?.header}</h6>
              <DonutChartLarge
                data={presentToEndCustomData}
                width={pieChartRenderSize}
                height={pieChartRenderSize}
                chartType={opexChartType}
                chartSection={AppConstant.OPERATING_TIME}
                removeTextTop={true}
              />
            </div>
          </div>
        </div>
      </div>
    </>
  )
};

export default AmountOfOperatingTime;
