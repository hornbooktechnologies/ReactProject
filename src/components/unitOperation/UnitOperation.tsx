import React, { useEffect, useState } from "react";
import { useTranslation } from 'react-i18next';
import { Kpi_005ApiRes, UnitOperationBaseModel } from "../../models/unitOperation/UnitOperationModel";
import { unitOperationApiMockData, unitOperationMockData } from "../../models/unitOperation/_mockData";
import LoadingError from "../common/LoadingError";
import PageHeader from "../common/PageHeader";
import Spinner from "../common/Spinner";
import AmountOfOperatingTime from "./amountOfOperatingTime/AmountOfOperatingTime";
import AverageSpread from "./averageSpread/AverageSpread";
import SpreadNumberOfCases from "./spreadNumberOfCases/SpreadNumberOfCases";
import { spreadOperationMockData } from "../../models/unitOperation/_mockData";
import NegativeOperationGrossMargin from "./NegativeOperationGrossMargin/NegativeGrossMarginTable";
import ReferenceSpecificationOfSpreadOperation from "./specificationSpread/ReferenceSpecificationOfSpreadOperation";
import SpecificationSpread from "./specificationSpread/SpecificationSpread";
import Top20DurationOfSpread from "./top20DurationOfSpread/Top20DurationOfSpread";
import { getCutoffFractionTime } from "../../utils/utils";
import { negOperationAvrgSpreadApiResponse, negOperationGrossmarginApiResponse, negOperationTimeApiResponse } from "../../utils/unitOperation/UnitOperationUtils";
import { NegativeOperationTimeModel} from "../../models/unitOperation/NegOperationTimeModel";
import { NegativeOperationAvgSpreadModel } from "../../models/unitOperation/NegOperationAvgSpreadModel.ts";
import { NegativeGrossMarginTableModel } from "../../models/unitOperation/NegOperationGrossmarginModel";
import "../../assets/scss/annualTarget/_annualstatus.scss";
import '../../assets/scss/unitOperation/unitOperation.scss';
import { DpmService } from "../../services/DpmServices";
import { ApiParamUtils } from "../../utils/apiUtils/ApiParamUtils";

const UnitOperation = () => {
  const { i18n } = useTranslation();
  const [unitOperationData, setUnitOperationData] = useState({} as UnitOperationBaseModel);
  const [negOperationGrossMarginData, setNegOperationGrossMargin] = useState({} as NegativeGrossMarginTableModel)
  const [negOperationTimeData, setNegOperationTime] = useState([] as NegativeOperationTimeModel)
  const [negOperationAvrgSpreadData, setNegOperationAvrgSpread] = useState([] as NegativeOperationAvgSpreadModel)
  const [unitOperationApiData, setUnitOperationApiData] = useState({});
  const [errorThrown, seterrorThrown] = useState(null)
  const [kpiResponse, setKpiResponse] = useState(false)
  const [time, setNoDataDate] = useState(
    getCutoffFractionTime()
  )

  const noDataHandler = (val: number, selectValue: string) => {
    setNoDataDate(val)
  }
    
  useEffect(() => {
    DpmService.fetchKPI_005Data(ApiParamUtils.plantId(),ApiParamUtils.unitCode()) 
    .then((res)=>{
      res=res.data
      if (res.Success) {
        setKpiResponse(res.Success)
        let kpiResponse = res.Data as Kpi_005ApiRes;
        let mappedNegGrossMarginResponse = negOperationGrossmarginApiResponse( kpiResponse);
        let mappedNegOperationTimeResponse = negOperationTimeApiResponse(i18n, kpiResponse);
        let mappedNegAverageSpread = negOperationAvrgSpreadApiResponse(i18n , kpiResponse);
        setNegOperationGrossMargin(mappedNegGrossMarginResponse)
        setNegOperationTime(mappedNegOperationTimeResponse)
        setNegOperationAvrgSpread(mappedNegAverageSpread)
       }
    })
    .catch((error) => {
      setKpiResponse(false)
      seterrorThrown(error)
     })
    setUnitOperationData(unitOperationMockData)
    setUnitOperationApiData(unitOperationApiMockData)
  }, [i18n])

  if (kpiResponse === true && Object.entries(unitOperationData && unitOperationApiData).length > 0) { 
    return (
      <>
        <PageHeader
          title={i18n.t('UNIT_OPERATION.TITLE')}
          description={i18n.t('UNIT_OPERATION.DESCRIPTION')}
        />
        <NegativeOperationGrossMargin negativeGrossMargin = {negOperationGrossMarginData}/>
        <AmountOfOperatingTime negOperationTime = {negOperationTimeData} />
        <AverageSpread negOperationAvrgSpread = {negOperationAvrgSpreadData} />
        <SpecificationSpread spreadOperation={spreadOperationMockData} time = {time} noDataHandler={noDataHandler}/>
        <ReferenceSpecificationOfSpreadOperation {...unitOperationData.referenceSpecification} />
        <SpreadNumberOfCases spreadBarData={unitOperationData.spreadBarChart}/>
        <Top20DurationOfSpread {...unitOperationData.top20DurationOfSpread} />
      </>
    );
  } else if (errorThrown != null) {
    return <LoadingError />;
  } else {
    return <Spinner />;
  }
};
export default UnitOperation;
