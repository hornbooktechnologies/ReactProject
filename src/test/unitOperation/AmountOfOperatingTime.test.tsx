import React from "react";
import renderer from "react-test-renderer";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
import AmountOfOperatingTime from './../../components/unitOperation/amountOfOperatingTime/AmountOfOperatingTime';
import { unitOperationMockData } from "../../models/unitOperation/_mockData";

describe("Component: AmountOfOperatingTime", () => {
  it("AmountOfOperatingTime renders correctly", () => {
    const unitOperationData = unitOperationMockData;
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <AmountOfOperatingTime {...unitOperationData.amountOfOperatingTime}/>
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
