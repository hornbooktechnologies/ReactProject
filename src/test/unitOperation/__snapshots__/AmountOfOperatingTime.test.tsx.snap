// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AmountOfOperatingTime AmountOfOperatingTime renders correctly 1`] = `
<div>
  <h5
    className="mb-24 mt-24 font-18 uk-text-medium"
  >
    UNIT_OPERATION.AMOUNT_OPERATING_SPREAD
  </h5>
  <div
    className="uk-grid uk-child-width-1-3@s uk-child-width-1-3@m"
    uk-grid="true"
  >
    <div
      className="uk-first-column"
    >
      <div
        className="uk-card"
      >
        <h6
          className="font-16"
        />
        <div
          className="uk-flex"
        >
          <div
            className="uk-flex-left uk-width-1-2 uk-text-center"
          >
            <label
              className="font-14"
            />
            <p
              className="mb-0 mt-5 font-14"
            >
              <span
                className=""
              >
                <strong>
                  
                  <span
                    className="seriesValue"
                  >
                    0
                  </span>
                  
                </strong>
              </span>
            </p>
          </div>
          <div
            className="uk-flex-left uk-width-1-2 uk-text-center"
          >
            <label
              className="font-14"
            />
            <p
              className="mb-0 mt-5 font-14"
            >
              <span
                className=""
              >
                <strong>
                  
                  <span
                    className="seriesValue"
                  >
                    0
                  </span>
                  
                </strong>
              </span>
            </p>
          </div>
        </div>
        <div
          className="uk-flex uk-flex-center undefined"
        >
          <div
            className="mt-30"
          >
            <div
              className="uk-flex uk-flex-center"
            >
              <div
                className="chart-continer"
                style={
                  Object {
                    "height": 190,
                    "width": 190,
                  }
                }
              >
                <div
                  className="dount-chart-container"
                >
                  <div />
                </div>
                <div
                  className=" donutmiddleData"
                >
                  <div
                    className="uk-text-center thumb-font"
                  >
                    <div
                      className="jera-like-font"
                    >
                      <i
                        className="jera color-green-60 jera-like-1"
                      />
                    </div>
                    <div
                      className="font-prvalue color-green-60"
                    >
                      <span>
                        0
                      </span>
                      <span
                        className="font-12 vertical-top"
                      >
                        %
                      </span>
                    </div>
                    <div>
                      <span
                        className="font-12 totalRatio"
                      >
                        CHARTS.DONUT.TOOLTIP_HEAD
                      </span>
                      <span
                        className="uk-tooltip-cutom tooltip-text"
                      >
                        <i
                          className="jera-info-circle icn-16 color-blue-70 top-2 ml-3"
                        />
                        <div
                          className="uk-tooltip uk-animation-scale-up uk-animation-center uk-transdorm-origin-center-left uk-togglabe-enter uk-tooltip-bottom uk-tooltip-bottom-center uk-transform-origin-center-left uk-text-left font-10"
                        >
                          <span
                            className="color-grey-blue-80 uk-display-block"
                          >
                            <strong
                              className="color-grey-blue-95"
                            >
                              CHARTS.DONUT.TOOLTIP_MAIN_TEXT
                            </strong>
                              
                            CHARTS.DONUT.TOOLTIP_SUB_TEXT
                          </span>
                          <span
                            className="color-red-60 mt-15 uk-display-block"
                          >
                            CHARTS.DONUT.EQUATION
                          </span>
                          <span
                            className="color-grey-blue-80 uk-display-block"
                          >
                            CHARTS.DONUT.EQUATION_LHS
                             = Y (
                            ) / X (
                            )
                          </span>
                        </div>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="media-mt-30 uk-first-column"
    >
      <div
        className="uk-card"
      >
        <h6
          className="font-16"
        >
          
        </h6>
        <div
          className="uk-flex"
        >
          <div
            className="uk-flex-left uk-width-1-2 uk-text-center"
          >
            <label
              className="font-14"
            />
            <p
              className="mb-0 mt-5 font-14"
            >
              <span
                className=""
              >
                <strong>
                  
                  <span
                    className="seriesValue"
                  >
                    0
                  </span>
                  
                </strong>
              </span>
            </p>
          </div>
          <div
            className="uk-flex-left uk-width-1-2 uk-text-center"
          >
            <label
              className="font-14"
            />
            <p
              className="mb-0 mt-5 font-14"
            >
              <span
                className=""
              >
                <strong>
                  
                  <span
                    className="seriesValue"
                  >
                    0
                  </span>
                  
                </strong>
              </span>
            </p>
          </div>
        </div>
        <div
          className="uk-flex uk-flex-center undefined"
        >
          <div
            className="mt-30"
          >
            <div
              className="uk-flex uk-flex-center"
            >
              <div
                className="chart-continer"
                style={
                  Object {
                    "height": 190,
                    "width": 190,
                  }
                }
              >
                <div
                  className="dount-chart-container"
                >
                  <div />
                </div>
                <div
                  className=" donutmiddleData"
                >
                  <div
                    className="uk-text-center thumb-font"
                  >
                    <div
                      className="jera-like-font"
                    >
                      <i
                        className="jera color-green-60 jera-like-1"
                      />
                    </div>
                    <div
                      className="font-prvalue color-green-60"
                    >
                      <span>
                        0
                      </span>
                      <span
                        className="font-12 vertical-top"
                      >
                        %
                      </span>
                    </div>
                    <div>
                      <span
                        className="font-12 totalRatio"
                      >
                        CHARTS.DONUT.TOOLTIP_HEAD
                      </span>
                      <span
                        className="uk-tooltip-cutom tooltip-text"
                      >
                        <i
                          className="jera-info-circle icn-16 color-blue-70 top-2 ml-3"
                        />
                        <div
                          className="uk-tooltip uk-animation-scale-up uk-animation-center uk-transdorm-origin-center-left uk-togglabe-enter uk-tooltip-bottom uk-tooltip-bottom-center uk-transform-origin-center-left uk-text-left font-10"
                        >
                          <span
                            className="color-grey-blue-80 uk-display-block"
                          >
                            <strong
                              className="color-grey-blue-95"
                            >
                              CHARTS.DONUT.TOOLTIP_MAIN_TEXT
                            </strong>
                              
                            CHARTS.DONUT.TOOLTIP_SUB_TEXT
                          </span>
                          <span
                            className="color-red-60 mt-15 uk-display-block"
                          >
                            CHARTS.DONUT.EQUATION
                          </span>
                          <span
                            className="color-grey-blue-80 uk-display-block"
                          >
                            CHARTS.DONUT.EQUATION_LHS
                             = Y (
                            ) / X (
                            )
                          </span>
                        </div>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className=" media-mt-30 uk-first-column"
    >
      <div
        className="uk-card"
      >
        <h6
          className="font-16"
        >
          
        </h6>
        <div
          className="uk-flex"
        >
          <div
            className="uk-flex-left uk-width-1-2 uk-text-center"
          >
            <label
              className="font-14"
            />
            <p
              className="mb-0 mt-5 font-14"
            >
              <span
                className=""
              >
                <strong>
                  
                  <span
                    className="seriesValue"
                  >
                    0
                  </span>
                  
                </strong>
              </span>
            </p>
          </div>
          <div
            className="uk-flex-left uk-width-1-2 uk-text-center"
          >
            <label
              className="font-14"
            />
            <p
              className="mb-0 mt-5 font-14"
            >
              <span
                className=""
              >
                <strong>
                  
                  <span
                    className="seriesValue"
                  >
                    0
                  </span>
                  
                </strong>
              </span>
            </p>
          </div>
        </div>
        <div
          className="uk-flex uk-flex-center undefined"
        >
          <div
            className="mt-30"
          >
            <div
              className="uk-flex uk-flex-center"
            >
              <div
                className="chart-continer"
                style={
                  Object {
                    "height": 190,
                    "width": 190,
                  }
                }
              >
                <div
                  className="dount-chart-container"
                >
                  <div />
                </div>
                <div
                  className=" donutmiddleData"
                >
                  <div
                    className="uk-text-center thumb-font"
                  >
                    <div
                      className="jera-like-font"
                    >
                      <i
                        className="jera color-green-60 jera-like-1"
                      />
                    </div>
                    <div
                      className="font-prvalue color-green-60"
                    >
                      <span>
                        0
                      </span>
                      <span
                        className="font-12 vertical-top"
                      >
                        %
                      </span>
                    </div>
                    <div>
                      <span
                        className="font-12 totalRatio"
                      >
                        CHARTS.DONUT.TOOLTIP_HEAD
                      </span>
                      <span
                        className="uk-tooltip-cutom tooltip-text"
                      >
                        <i
                          className="jera-info-circle icn-16 color-blue-70 top-2 ml-3"
                        />
                        <div
                          className="uk-tooltip uk-animation-scale-up uk-animation-center uk-transdorm-origin-center-left uk-togglabe-enter uk-tooltip-bottom uk-tooltip-bottom-center uk-transform-origin-center-left uk-text-left font-10"
                        >
                          <span
                            className="color-grey-blue-80 uk-display-block"
                          >
                            <strong
                              className="color-grey-blue-95"
                            >
                              CHARTS.DONUT.TOOLTIP_MAIN_TEXT
                            </strong>
                              
                            CHARTS.DONUT.TOOLTIP_SUB_TEXT
                          </span>
                          <span
                            className="color-red-60 mt-15 uk-display-block"
                          >
                            CHARTS.DONUT.EQUATION
                          </span>
                          <span
                            className="color-grey-blue-80 uk-display-block"
                          >
                            CHARTS.DONUT.EQUATION_LHS
                             = Y (
                            ) / X (
                            )
                          </span>
                        </div>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
